packages/lib-classifier/src/plugins/drawingTools/components/Polygon/Polygon.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { observer } from 'mobx-react'
+import styled from 'styled-components'
+import UndoButton from '../UndoButton'
+import DragHandle from '../DragHandle'
+
+const GuideLine = styled.line`
+  pointer-events: none;
+`
+
+const RADIUS = 3
+const STROKE_WIDTH = 3
+const GUIDELINE_STROKE_WIDTH = 2
+const GRAB_STROKE_WIDTH = 6
+
+function Polygon({ active, mark, scale }) {
+  const {
+    path,
+    points,
+    initialPoint,
+    lastPoint,
+    finished,
+    guideLineX,
+    guideLineY
+  } = mark
+
+  const radius = RADIUS / scale
+  const strokeWidth = STROKE_WIDTH / scale
+  const guideLineStrokeWidth = GUIDELINE_STROKE_WIDTH / scale
+  const grabStrokeWidth = GRAB_STROKE_WIDTH / scale
+
+  function onHandleDrag(coords, i) {
+    mark.setCoordinates(coords, i)
+  }
+
+  function onUndoDrawing() {
+    mark.continueDrawing()
+    mark.shortenPath()
+  }
+
+  function handleClosePolygon(i) {
```

Comment:
In Polygon component, `handleClosePolygon()` is only used for an initial point's `onPointerDown()` event. There’s a check in `handleClosePolygon()` for index === 0, but there’s no case where `handleClosePolygon()` is called with an index other than 0. I think `handleClosePolygon()` could simply call `mark.finish()` and doesn't need to take a parameter.